bc_obps/reporting/service/compliance_service.py

```diff
@@ -235,17 +238,28 @@ def get_calculated_compliance_data(cls, report_version_id: int) -> ComplianceDat
             )
             allocated_reporting_only = ComplianceService.get_reporting_only_allocated(report_version_id, rp.product_id)
             allocated_for_compliance = allocated - allocated_reporting_only
-            allocated_for_compliance_2024 = (
-                Decimal(0)
-                if Decimal(production_totals["annual_amount"]) == 0
-                else (
-                    (allocated_for_compliance / Decimal(production_totals["annual_amount"]))
-                    * Decimal(production_totals["apr_dec"])
+            # If this compliance period is 2024, use Apr-Dec production for allocations and limits.
+            # Otherwise use full-year production and full-year allocated emissions.
+            if use_apr_dec:
+                allocated_for_compliance_2024 = (
+                    Decimal(0)
+                    if Decimal(production_totals["annual_amount"]) == 0
+                    else (
+                        (allocated_for_compliance / Decimal(production_totals["annual_amount"]))
+                        * Decimal(production_totals["apr_dec"])
+                    )
                 )
-            )
+                production_for_limit = Decimal(production_totals["apr_dec"])
+                allocated_compliance_emissions_value = round(allocated_for_compliance_2024, 4)
+            else:
+                # Not 2024: ignore apr-dec window and use full-year values
+                allocated_for_compliance_2024 = Decimal(0)
+                production_for_limit = Decimal(production_totals["annual_amount"])
+                allocated_compliance_emissions_value = round(allocated_for_compliance, 4)
+
             product_emission_limit = ComplianceService.calculate_product_emission_limit(
                 pwaei=ei,
-                apr_dec_production=Decimal(production_totals["apr_dec"]),
+                apr_dec_production=production_for_limit,
```

Comment:
the parameter should be renamed to just `production_amount` or `production_for_emission_limit` or something like that